### ISSUE TITLE
Could not get unknown property 'javaToolchains' for task

### ISSUE THREAD

**Issue by nori-flutter** [fluttercommunity/plus_plugins#324]
## System info

Issue occurs on: Android only
Plugin name: connectivity_plus
Plugin version: 1.0.4
Flutter doctor output:

```
flutter doctor -v 
[✓] Flutter (Channel stable, 2.2.2, on macOS 11.4 20F71 darwin-x64, locale en)
    • Flutter version 2.2.2 at /Users/user/flutter
    • Framework revision d79295af24 (2 weeks ago), 2021-06-11 08:56:01 -0700
    • Engine revision 91c9fc8fe0
    • Dart version 2.13.3

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.3)
    • Android SDK at /Users/user/Library/Android/sdk
    • Platform android-S, build-tools 30.0.3
    • ANDROID_HOME = /Users/user/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.5.1, Build version 12E507
    • CocoaPods version 1.10.1

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.8+10-b944.6916264)

[✓] VS Code (version 1.57.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.23.0

[✓] Connected device (4 available)
    • Lenovo TB 7305F (mobile) • HA116W57                                 • android-arm    • Android 9 (API 28)
    • macOS (desktop)          • macos                                    • darwin-x64     • macOS 11.4 20F71 darwin-x64
    • Chrome (web)             • chrome                                   • web-javascript • Google Chrome 91.0.4472.114

• No issues found!
```

## Steps to Reproduce
Everything was okay with connectivity_plus-1.0.3
1. Upgraded connectivity_plus-1.0.3 to connectivity_plus-1.0.4
2. Build my app
3. Build failed

## Logs
```
Running Gradle task 'assembleDebug'...

FAILURE: Build completed with 2 failures.

1: Task failed with an exception.
-----------
* Where:
Build file '/Users/user/flutter/.pub-cache/hosted/pub.dartlang.org/connectivity_plus-1.0.4/android/build.gradle' line: 22

* What went wrong:
A problem occurred evaluating project ':connectivity_plus'.
> Could not get unknown property 'javaToolchains' for task ':app:compileDebugJavaWithJavac' of type org.gradle.api.tasks.compile.JavaCompile.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

2: Task failed with an exception.
-----------
* Where:
Script '/Users/user/flutter/packages/flutter_tools/gradle/flutter.gradle' line: 346

* What went wrong:
A problem occurred configuring project ':connectivity_plus'.
> Could not get unknown property 'android' for project ':connectivity_plus' of type org.gradle.api.Project.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
==============================================================================

* Get more help at https://help.gradle.org

BUILD FAILED in 2s
Exception: Gradle task assembleDebug failed with exit code 1
```

I temporally commented out the below section in connectivity_plus-1.0.4/android/build.gradle to make my urgent app build.
```
//    tasks.withType(JavaCompile).configureEach {
//    javaCompiler = javaToolchains.compilerFor {
//      languageVersion = JavaLanguageVersion.of(8)
//    }
//  }
```

Here is Java version on my Mac (MacOS Big Sir 11.4)
```
java -version
java version "14.0.2" 2020-07-14
Java(TM) SE Runtime Environment (build 14.0.2+12-46)
Java HotSpot(TM) 64-Bit Server VM (build 14.0.2+12-46, mixed mode, sharing)
```

Please let me know how to resolve "javaToolchains" and JavaLanguageVersion.of(8) in the connectivity_plus-1.0.4/android/build.gradle

**Comment by TusharGupta05** [comment]
I am also facing this issue

**Comment by mvquezm** [comment]
Same issue

**Comment by pawel-gasiorowski-avenga** [comment]
I have the same issue, but the issue occurs on DevOps Pipeline when building the app. Before adding connectivity, it was working fine. 

`Running Gradle task 'assembleRelease'...                        

FAILURE: Build failed with an exception.

* Where:
Build file '/Users/runner/hostedtoolcache/Flutter/2.3.0-16.0.pre-dev/macos/flutter/.pub-cache/hosted/pub.dartlang.org/connectivity_plus-1.0.4/android/build.gradle' line: 22

* What went wrong:
A problem occurred evaluating project ':connectivity_plus'.
> Could not get unknown property 'javaToolchains' for task ':app:compileDebugAndroidTestJavaWithJavac' of type com.android.build.gradle.tasks.AndroidJavaCompile.

* Try:
Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.

* Get more help at https://help.gradle.org

BUILD FAILED in 1m 2s
Running Gradle task 'assembleRelease'...                           65.2s
Gradle task assembleRelease failed with exit code 1
`

**Comment by benitogonzalezh** [comment]
Solved in #321

**Comment by nori-flutter** [comment]
> Solved in #321

Thank you for the info. Yes, it's resolved by upgrading the gradle

I have applied the below changes and the build success now:

android/build.gradle
```
        classpath 'com.android.tools.build:gradle:3.6.4'
to
        classpath 'com.android.tools.build:gradle:4.2.1'
```

android/gradle/wrapper/gradle-wrapper.properties
```
distributionUrl=https\://services.gradle.org/distributions/gradle-6.1.1-all.zip
to
distributionUrl=https\://services.gradle.org/distributions/gradle-6.8-all.zip
```
